File: ios/SiriApp/Models/ChatMessage.swift
//
//  ChatMessage.swift
//  SiriApp
//
//  Ported from Android: ChatMessage.kt
//

import Foundation

struct ChatMessage: Identifiable, Codable {
    let id: String
    let role: Role
    let content: String
    let timestamp: Date

    enum Role: String, Codable {
        case user
        case assistant
        case system

        var value: String { rawValue }

        static func fromValue(_ value: String) -> Role {
            Role(rawValue: value) ?? .user
        }
    }

    init(id: String = UUID().uuidString,
         role: Role,
         content: String,
         timestamp: Date = Date()) {
        self.id = id
        self.role = role
        self.content = content
        self.timestamp = timestamp
    }
}

File: app/src/main/java/com/rd/siri/model/ChatMessage.kt
package com.rd.siri.model

import java.util.UUID

data class ChatMessage(
    val id: String = UUID.randomUUID().toString(),
    val role: Role,
    val content: String,
    val timestamp: Long = System.currentTimeMillis()
) {
    enum class Role(val value: String) {
        USER("user"),
        ASSISTANT("assistant"),
        SYSTEM("system");

        companion object {
            fun fromValue(value: String): Role =
                entries.firstOrNull { it.value == value } ?: USER
        }
    }
}
